lib/pf/provisioner.pm

```diff
@@ -190,7 +194,7 @@ sub matchOS {
 
 sub match {
     my ($self, $os, $node_attributes) = @_;
-    return $self->matchOS($os) && $self->matchCategory($node_attributes);
+    return $self->matchOS($node_attributes) && $self->matchCategory($node_attributes);
```

Comment:
change this to `$self->matchCategory($node_attributes) && $self->matchOS($node_attributes)` to avoid unnecessary look ups to fingerbank if matchCategory does not match.